src/Helpers/useUrlParams.js

```diff
@@ -0,0 +1,34 @@
+import { useCallback } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+// TODO DRY:similar to constructParameters
```

Comment:
Maybe it is time to finally to implement this `TODO`. Can we try to consolidate this function with `constructParameters` and get rid of the duplicate code?